src/NServiceBus.Core/Pipeline/MutateInstanceMessage/MutateIncomingMessageBehavior.cs

```diff
@@ -15,7 +15,9 @@ class MutateIncomingMessageBehavior : Behavior<IIncomingLogicalMessageContext>
             var mutatorContext = new MutateIncomingMessageContext(current, context.Headers);
             foreach (var mutator in context.Builder.BuildAll<IMutateIncomingMessages>())
             {
-                await mutator.MutateIncoming(mutatorContext).ConfigureAwait(false);
+                await mutator.MutateIncoming(mutatorContext)
```

Comment:
`MethodInfo` was just the reference from the reflection we already do, but saving all of that isn't necessary. Replaced with just saving the message `Type` and method Name `string` value